src/test/java/io/openliberty/tools/intellij/it/GradleSingleModMPProjectTest.java

```diff
@@ -88,131 +35,18 @@ public static void setup() {
-    /**
-     * Returns the expected HTTP response payload associated with the single module
-     * MicroProfile project.
-     *
-     * @return The expected HTTP response payload associated with the single module
-     * MicroProfile project.
-     */
-    @Override
-    public String getSmMPProjOutput() {
-        return SM_MP_PROJECT_OUTPUT;
-    }
-
-    /**
-     * Returns the port number associated with the single module MicroProfile project.
-     *
-     * @return The port number associated with the single module MicroProfile project.
-     */
-    @Override
-    public int getSmMpProjPort() {
-        return SM_MP_PROJECT_PORT;
-    }
-
-    /**
-     * Return the Resource URI associated with the single module MicroProfile project.
-     *
-     * @return The Resource URI associated with the single module MicroProfile project.
-     */
-    @Override
-    public String getSmMpProjResURI() {
-        return SM_MP_PROJECT_RES_URI;
-    }
-
-    /**
-     * Returns the name of the build file used by the project.
-     *
-     * @return The name of the build file used by the project.
-     */
-    @Override
-    public String getBuildFileName() {
-        return BUILD_FILE_NAME;
-    }
-
-    /**
-     * Returns the name of the custom action command used to open the build file.
-     *
-     * @return The name of the custom action command used to open the build file.
-     */
-    @Override
-    public String getBuildFileOpenCommand() {
-        return BUILD_FILE_OPEN_CMD;
-    }
-
-    /**
-     * Returns the custom start parameters to be used to start dev mode.
-     *
-     * @return The custom start parameters to be used to start dev mode.
-     */
-    @Override
-    public String getStartParams() {
-        return DEV_MODE_START_PARAMS;
-    }
-
-    /**
-     * Returns the custom start parameters for debugging to start dev mode.
-     *
-     * @return The custom start parameters for debugging to start dev mode.
-     */
-    @Override
-    public String getStartParamsDebugPort() {
-        return DEV_MODE_START_PARAMS_DEBUG;
-    }
-
-    /**
-     * Deletes test reports.
-     */
-    @Override
-    public void deleteTestReports() {
-        boolean testReportDeleted = TestUtils.deleteFile(TEST_REPORT_PATH);
-        Assertions.assertTrue(testReportDeleted, () -> "Test report file: " + TEST_REPORT_PATH + " was not be deleted.");
-    }
-
-    /**
-     * Validates that test reports were generated.
-     */
-    @Override
-    public void validateTestReportsExist() {
-        //TODO: rewrite validateTestReportExists() to accept one argument or to accept a null as the second argument
-        TestUtils.validateTestReportExists(TEST_REPORT_PATH, TEST_REPORT_PATH);
-    }
-
-    /**
-     * Returns the build type category (either MAVEN_TYPE or GRADLE_TYPE)
-     */
-    @Override
-    public BuildType getBuildCategory() {
-        return BUILD_CATEGORY;
+    GradleSingleModMPProjectTest() {
+        setProjectsDirPath(PROJECTS_PATH);
```

Comment:
It will be better to add some comments before each of these setting so that it makes it easy to quickly understand the purpose of each configuration ?

Reply:
I was hoping these attributes were self explanatory and the comments in the base class would remove any ambiguity. Also, I put the project path and name first and other attributes later to indicate the importance of each.

Reply:
Ok .Its fine then. Thanks!!@turkeylurkey @mrglavas.